community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/setup_network_storage.py

```diff
@@ -78,74 +94,46 @@ def resolve_network_storage(nodeset=None) -> List[NSMount]:
         except Exception:
             pass # external nodename, skip lookup
         else:
-            mounts.update(mounts_by_local(nodeset.network_storage))
-
-    return [lkp.normalize_ns_mount(mnt) for mnt in mounts.values()]
-
+            nodeset_ns = [lkp.normalize_ns_mount(m) for m in nodeset.network_storage]
+            mounts.update(mounts_by_local(nodeset_ns))
 
-def separate_external_internal_mounts(mounts):
-    """separate into cluster-external and internal mounts"""
+    return list(mounts.values())
 
-    def internal_mount(mount):
-        # NOTE: Valid Lustre server_ip can take the form of '<IP>@tcp'
-        server_ip = mount.server_ip.split("@")[0]
-        mount_addr = util.host_lookup(server_ip)
-        return mount_addr == lookup().control_host_addr
-
-    return separate(internal_mount, mounts)
 
+def is_controller_mount(mount) -> bool:
+    # NOTE: Valid Lustre server_ip can take the form of '<IP>@tcp'
+    server_ip = mount.server_ip.split("@")[0]
+    mount_addr = util.host_lookup(server_ip)
```

Comment:
nit: Took me a second to realize this is not a mount for the controller but rather a mount that is mounted to the controller (akin to internal mount). So the mounts that you separate in L115 are explicit the non-controller mounts (a.k.a external mounts in the old code).

Took me a bit to follow this so I wonder if keeping the name `internal_mount` would make it more obvious. Not really sure.